thirty-five-mm/app/models/film.rb

```diff
@@ -0,0 +1,21 @@
+class Film < ActiveRecord::Base
+  belongs_to :category
+  has_many :reviews
+  has_many :ratings, as: :ratable
+  has_many :comments, as: :commentable
+
+  validates :category, :title, presence: true
+
+
+  def self.search(search)
+  	if search
+  		Film.where(title: search).to_a
+  	else
+  		Film.all
+  	end
+  end
+
+	def no_of_stars
+    self.reviews.average('no_of_stars')
```

Comment:
It's not doing any harm, but you don't need self. here - it's only needed when you are writing an attribute.